src/Discord.Net.Interactions/InteractionService.cs

```diff
@@ -588,6 +588,58 @@ private bool RemoveModuleInternal (ModuleInfo moduleInfo)
             return true;
         }
 
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="ISlashCommandInteraction"/>.
+        /// </summary>
+        /// <param name="slashCommandInteraction">Interaction entity to perform the search with.</param>
+        /// <returns>
+        ///     The search result.
+        /// </returns>
+        public SearchResult<SlashCommandInfo> SearchSlashCommand(ISlashCommandInteraction slashCommandInteraction)
+            => _slashCommandMap.GetCommand(slashCommandInteraction.Data.GetCommandKeywords());
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IComponentInteraction"/>.
+        /// </summary>
+        /// <param name="componentInteraction">Interaction entity to perform the search with.</param>
+        /// <returns></returns>
+        public SearchResult<ComponentCommandInfo> SearchComponentCommand(IComponentInteraction componentInteraction)
+            => _componentCommandMap.GetCommand(componentInteraction.Data.CustomId);
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IUserCommandInteraction"/>.
+        /// </summary>
+        /// <param name="userCommandInteraction">Interaction entity to perform the search with.</param>
+        /// <returns>
+        ///     The search result.
+        /// </returns>
+        public SearchResult<ContextCommandInfo> SearchUserCommand(IUserCommandInteraction userCommandInteraction)
+            => _contextCommandMaps[ApplicationCommandType.User].GetCommand(userCommandInteraction.Data.Name);
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IMessageCommandInteraction"/>.
+        /// </summary>
+        /// <param name="messageCommandInteraction">Interaction entity to perform the search with.</param>
+        /// <returns>
+        ///     The search result.
+        /// </returns>
+        public SearchResult<ContextCommandInfo> SearchMessageCommand(IMessageCommandInteraction messageCommandInteraction)
+            => _contextCommandMaps[ApplicationCommandType.Message].GetCommand(messageCommandInteraction.Data.Name);
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IAutocompleteInteraction"/>.
```

Comment:
What does 'loaded' imply, maybe change to 'registered' to match the naming of other methods in the service?